### PR TITLE
 refactor commented_code_linter() to use xml_nodes_to_lints()

### LINKED ISSUE #1274: refactor `comment_linters()` to use xml_nodes_to_lints()

### PR DESCRIPTION
**Pull request by AshesITR**
fixes #1274

### CODE REVIEW

File: R/comment_linters.R [hunk 1]
```diff
@@ -54,32 +54,34 @@ commented_code_linter <- function() {
...
+    is_parsable <- which(vapply(code_candidates[, "code"], parsable, logical(1L)))
 
-    # TODO(michaelchirico): convert this to xml_nodes_to_lint()?
-    lapply(rownames(na.omit(code_candidates)), function(code_candidate) {
-      is_parsable <- parsable(code_candidates[code_candidate, "code"])
-      if (is_parsable) {
-        comment_node <- all_comment_nodes[[as.integer(code_candidate)]]
-        line_number <- as.integer(xml2::xml_attr(comment_node, "line1"))
-        column_offset <- as.integer(xml2::xml_attr(comment_node, "col1")) - 1L
+    lint_list <- xml_nodes_to_lints(
```
Comment (MichaelChirico) [inline R/comment_linters.R:59]: hmm I'd say no need to force xml_nodes_to_lint if it's not up to the task -- here may be such a case

Comment (AshesITR) [reply R/comment_linters.R:59]: I like that it abstracts away the boilerplate. The implementation looks much cleaner imo.

Comment (AshesITR) [reply R/comment_linters.R:59]: on tools/QC.R we lose 270ms (34 lints produced) but this is surely recovered by the optimizations to `xml_nodes_to_lints()` in #1284.

If you want, we merge that first and I do another performance comparison?

Comment (MichaelChirico) [reply R/comment_linters.R:59]: I'm not going to lose sleep over 270ms in one of the most unwieldy R scripts around :)

**Review by MichaelChirico — APPROVED** [review]

File: R/comment_linters.R (hunk 1 above)
Comment (AshesITR) [reply R/comment_linters.R:59]: True :) will hold off just out of scientific curiosity.

**Pull request merged**